docs/operate/reference/viam-server/debug-endpoints.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "Debug Endpoints"
+linkTitle: "Debug Endpoints"
+weight: 130
+type: "docs"
+description: "Advanced debugging endpoints available in viam-server for troubleshooting and development."
+date: "2025-05-19"
+---
+
+The RDK (Robot Development Kit) that powers `viam-server` includes several debug endpoints that expose server internals.
+These endpoints can be helpful during development or when troubleshooting complex issues:
+
+- [`pprof`](#pprof)
+- [`graph`](#graph)
+
+## `pprof`
+
+Setting `enable_web_profile` to `true` in your machine configuration allows you to visualize runtime profiling data in the localhost browser.
+The pprof tool provides valuable insights into your application's performance, including:
+
+- CPU usage
+- Memory allocation
+- Goroutine blocking
+- Execution tracing
+
+This information can be particularly useful when:
+- Diagnosing performance bottlenecks
+- Investigating memory leaks
+- Understanding resource usage patterns
+- Troubleshooting high CPU or memory usage
+
+For more detailed information on using pprof, refer to the [Go pprof documentation](https://pkg.go.dev/net/http/pprof).
+
+### Configuration
+
+To enable the pprof endpoints, set `enable_web_profile` to `true` in your machine configuration:
+
+1. Navigate to the **CONFIGURE** tab of your machine in the Viam app
+2. Click on the **...** menu next to your machine part in the left-hand menu
+3. Select **Edit JSON**
+4. Add the `enable_web_profile` field at the root level of your configuration:
```

Comment:
```suggestion
1. Navigate to the **CONFIGURE** tab of your machine in the Viam app
1. Click on the **...** menu next to your machine part in the left-hand menu
1. Select **Edit JSON**
1. Add the `enable_web_profile` field at the root level of your configuration:
```
[s] pet peeve of mine: never number ordered lists, always becomes a pain eventually. Maybe we can supplement the ai prompt with that instruction?